From e92876a5257bc762eb61c2f12f0338be493ab939 Mon Sep 17 00:00:00 2001
From: Rachel Menge <rachelmenge@microsoft.com>
Date: Wed, 29 May 2024 22:27:32 +0000
Subject: [PATCH] coreutils-9.4 uname patch for -i and -p

Original commit info:

Submitted by: William Immendorf <will.immendorf@gmail.com>
Date: 2010-05-08
Inital Package Version: 8.5
Origin: http://cvs.fedoraproject.org/viewvc/devel/coreutils/coreutils-8.2-uname-processortype.patch
Upstream Status: Rejected
Description: Fixes the output of uname's -i and -p parameters
---
 src/uname.c | 18 ++++++++++++++++--
 1 file changed, 16 insertions(+), 2 deletions(-)

diff --git a/src/uname.c b/src/uname.c
index 883b9a4..78641bf 100644
--- a/src/uname.c
+++ b/src/uname.c
@@ -313,7 +313,7 @@ main (int argc, char **argv)
 
   if (toprint & PRINT_PROCESSOR)
     {
-      char const *element = unknown;
+      char *element = unknown;
 #ifdef __APPLE__
 # if defined __arm__ || defined __arm64__
       element = "arm";
@@ -330,6 +330,12 @@ main (int argc, char **argv)
           if (0 <= sysinfo (SI_ARCHITECTURE, processor, sizeof processor))
             element = processor;
         }
+#else
+      {
+	struct utsname u;
+	uname(&u);
+	element = u.machine;
+      }
 #endif
 #ifdef UNAME_PROCESSOR
       if (element == unknown)
@@ -347,7 +353,7 @@ main (int argc, char **argv)
 
   if (toprint & PRINT_HARDWARE_PLATFORM)
     {
-      char const *element = unknown;
+      char *element = unknown;
 #if HAVE_SYSINFO && defined SI_PLATFORM
       {
         static char hardware_platform[257];
@@ -355,6 +361,14 @@ main (int argc, char **argv)
                           hardware_platform, sizeof hardware_platform))
           element = hardware_platform;
       }
+#else
+      {
+	struct utsname u;
+	uname(&u);
+	element = u.machine;
+	if(strlen(element)==4 && element[0]=='i' && element[2]=='8' && element[3]=='6')
+		element[1]='3';
+      }
 #endif
 #ifdef UNAME_HARDWARE_PLATFORM
       if (element == unknown)
-- 
2.34.1

